pkg/gateway/middleware.go

```diff
@@ -175,7 +176,7 @@ func OperationLookupHandler(next http.Handler) http.Handler {
 		ctx := req.Context()
 		o := ctx.Value(ContextKeyOperation).(*operations.Operation)
 		repoID := ctx.Value(ContextKeyRepositoryID).(string)
-		var operationID operations.OperationID
+		operationID := operations.OperationIDOperationNotFound
```

Comment:
I think we need to update the structure (o.OperationID) at this point as we do return in case of an error.

Reply:
removed `operationID` entirely :-)